classes/userGroup/Repository.php

```diff
@@ -590,12 +592,38 @@ public function getMastheadUserIdsByRoleIds(array $mastheadRoles, int $contextId
             $users = UserUserGroup::query()
                 ->withContextId($contextId)
                 ->withUserGroupIds($mastheadRoleIds)
-                ->withUserUserGroupStatus($userUserGroupStatus->value)
-                ->withUserUserGroupMastheadStatus(UserUserGroupMastheadStatus::STATUS_ON->value)
+                ->whereHas(
+                    'userGroup',
+                    fn (\Illuminate\Database\Eloquent\Builder $query) => $query->withUserUserGroupStatus($userUserGroupStatus->value)
+                )
+                ->withMasthead()
                 ->orderBy('user_groups.role_id', 'asc')
                 ->join('user_groups', 'user_user_groups.user_group_id', '=', 'user_groups.user_group_id')
                 ->join('users', 'user_user_groups.user_id', '=', 'users.user_id')
-                ->orderBy('users.family_name', 'asc')
+                ->orderBy(
+                    fn (\Illuminate\Database\Query\Builder $query) => $query
+                        ->fromSub(
+                            fn($query) => $query->from(null)->selectRaw(0),
+                            'placeholder'
+                        )
+                        ->leftJoin(
+                            'user_settings AS l',
+                            fn (JoinClause $join) => $join
+                                ->on('user_user_groups.user_id', '=', 'l.user_id')
+                                ->where('l.setting_name', 'family_name')
+                                ->where('l.locale', Locale::getLocale())
+                        )
+                        ->leftJoin(
+                            'user_settings AS p',
+                            fn (JoinClause $join) => $join
+                                ->on('user_user_groups.user_id', '=', 'p.user_id')
+                                ->where('p.setting_name', 'family_name')
+                                ->where('p.locale', Locale::getPrimaryLocale())
```

Comment:
I think Locale::getPrimaryLocale() would return context primary locale if possible, right? The user multilingual data do not necessarily exist in the context primary locale -- the fall back locale for required users data is the site primary locale.
Also, I think we need to consider the given_name as well i.e. the logic should be the same as in the user Collector class, because it can be that there is no family name, so we should in that case consider/order by given name (that is required and thus always there at least in the site primary locale).
I hope I understand the code correctly... :-)

Reply:
Hmmm... here https://github.com/pkp/pkp-lib/pull/10667/files#diff-c8cb812ae82cc024166167807c2d42818fe8c33bc61dc597835f7dbf2d7c91b2, the user collector did not change, so maybe to use the old part of the code here, that was changed/replaced in this PR, i.e. to user the user collector to get the users? :thinking:

Reply:
Do I understand it correctly, that here both family but then also given name will be considered here: https://github.com/pkp/pkp-lib/blob/main/classes/user/Collector.php#L779-L814 -- which was used earlier... ?

Reply:
Right, it does consider that . I have update the PR and revert back the changes to utilize the user collector features .